File: src/main/java/org/kite9/diagram/visualization/planarization/Planarizer.java
package org.kite9.diagram.visualization.planarization;

import org.kite9.diagram.adl.Diagram;

/**
 * Interface for creating a Planarization of a set of diagram attr suitable for further layout.
 * 
 * @author robmoffat
 *
 */
public interface Planarizer {

	public Planarization planarize(Diagram d);
	
}

File: src/main/kotlin/org/kite9/diagram/visualization/planarization/Planarizer.kt
package org.kite9.diagram.visualization.planarization

import org.kite9.diagram.model.Diagram

/**
 * Interface for creating a Planarization of a set of diagram attr suitable for further layout.
 *
 * @author robmoffat
 */
interface Planarizer {
    fun planarize(d: Diagram?): Planarization?
}